### PR TITLE
add .open for short-lived reporting that does not leave sockets around

### PR DESCRIPTION
**Pull request by grosser**
@gmmeyer @adammw

### CODE REVIEW

**Comment by grosser** [comment]
@gmmeyer

File: lib/datadog/statsd.rb [hunk 1]
```diff
@@ -77,7 +77,7 @@ def write(message)
 
       # Close the underlying socket
       def close
-        @socket.close
+        @socket && @socket.close
```
Comment (gmmeyer) [inline lib/datadog/statsd.rb]: 🍰

**Comment by grosser** [comment]
how about a v4.0.0 release ? :D

**Comment by gmmeyer** [comment]
@grosser we're gonna release soon! We just need to find a little time to do it!

**Pull request merged**